app/presenters/timeline_events/show_presenter.rb

```diff
@@ -11,27 +11,54 @@ def page_title
     end
 
     def markdown_prop(markdown)
-      { markdown: markdown, profile: 'permissive' }.to_json
+      { markdown: markdown, profile: "permissive" }.to_json
     end
 
     def render_markdown?(item)
-      %w[longText shortText link].include?(item['kind'])
+      %w[longText shortText].include?(item["kind"])
     end
 
     def files?(item)
-      item['kind'] == 'files'
+      item["kind"] == "files"
     end
 
     def audio?(item)
-      item['kind'] == 'audio'
+      item["kind"] == "audio"
+    end
+
+    def link?(item)
+      item["kind"] == "link"
     end
 
     def audio_file(item)
-      TimelineEventFile.find(item['result'])
+      TimelineEventFile.find(item["result"])
     end
 
     def files(item)
-      TimelineEventFile.where(id: item['result'])
+      TimelineEventFile.where(id: item["result"])
+    end
+
+    def submission_from
+      if @submission.founder.team.present? && @submission.team_submission?
+        @submission.founder.team.name
+      else
+        @submission.founder.name
+      end
+    end
```

Comment:
@kaisersakhi did you check what we are doing in the review dashboard? 

Consider this scenario, _John_ was part of _team_1_, and the team submitted a team target; he was later moved to another `team_2`. 

What will appear on the submission when the coach visits the page? (Assume _John_ was the 1st in the list).

Reply:
I see the problem with my approach.
In the review dashboard, we don't indicate if a submission is from a team when all students doesn't belong to the same tame, we simply show all team member names.
@bodhish  Should I do the same? I.e, show all student names if one members team has changed?

Reply:
Why not show all the students in all cases?

Reply:
I think that is the best solution to overcome *team change* edge case.

Reply:
The design doesn't work when we have more than one student; check with @irajsuhail 

Also show the team name when he has a team linked